pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -16,13 +18,14 @@ import (
 	"github.com/grafana/grafana/pkg/util"
 )
 
-func NewStore(sql db.DB, features featuremgmt.FeatureToggles) *store {
-	return &store{sql, features}
+func NewStore(sql db.DB, features featuremgmt.FeatureToggles, inMemoryActionSets *InMemoryActionSets) *store {
+	return &store{sql, features, *inMemoryActionSets}
 }
 
 type store struct {
-	sql      db.DB
-	features featuremgmt.FeatureToggles
+	sql                db.DB
+	features           featuremgmt.FeatureToggles
+	inMemoryActionSets InMemoryActionSets
```

Comment:
nit: I don't think we care about how action sets are implemented when we use them as a dependency. As in, we don't care if it's in memory or in a DB. Do you think it makes sense creating an interface around action sets, and using that (similarly as we do for other services/stores)?

Reply:
we did make a interface for the POC, but we decided it would be a bit too much for now. But perhaps we should introduce it again?

Reply:
https://github.com/grafana/grafana/pull/85662#discussion_r1555512885